client/src/components/HeaderSettings/HeaderSettings.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC, useState } from 'react'
+import Button from '@material-ui/core/Button'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import { createStyles, makeStyles } from '@material-ui/core'
+import { Theme } from '@material-ui/core/styles'
+import AuthModal from '../AuthModal/AuthModal'
+import SwitchSystem from '../Switch/SwitchSystem'
+import { useTypedSelector } from '../../hooks/useTypedSelector'
+import { useActions } from '../../hooks/useActions'
+import {
+    selectAuthIsLoading,
+    selectAuthStatus,
+} from '../../store/reducers/auth/selectors'
+
+const useStylesHeader = makeStyles((theme: Theme) =>
+    createStyles({
+        headerSettings: {
+            display: 'flex',
+            alignItems: 'center',
+            [theme.breakpoints.down('md')]: {
+                flexDirection: 'column',
+                minWidth: 90,
+            },
+        },
+    })
+)
+
+const HeaderSettings: FC = () => {
+    const classes = useStylesHeader()
+    const isAuth = useTypedSelector(selectAuthStatus)
+    const isLoading = useTypedSelector(selectAuthIsLoading)
+    const { logout } = useActions()
+    const [visibleModal, setVisibleModal] = useState(false)
+
+    const handleClickOpenModal = (): void => {
+        if (isLoading) {
+            return
```

Comment:
it's better to disable button while it's waiting

Reply:
corrected